modules/gapi/include/opencv2/gapi/stereo.hpp

```diff
@@ -0,0 +1,78 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distereoibution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#ifndef OPENCV_GAPI_STEREO_HPP
+#define OPENCV_GAPI_STEREO_HPP
+
+#include <opencv2/gapi/gmat.hpp>
+#include <opencv2/gapi/gscalar.hpp>
+#include <opencv2/gapi/gkernel.hpp>
+
+namespace cv {
+
+namespace gapi {
+namespace ster {
+
+enum class OUTPUT_FORMAT {
+    disparity,
+    depths
+};
+
+struct StereoSetup {
+    int numDisparities;
+    int blockSize;
+    double baseline;
+    double focus;
+};
+
+G_TYPED_KERNEL(GStereo, <GMat(GMat, GMat, const cv::gapi::ster::OUTPUT_FORMAT, const int)>, "org.opencv.stereo") {
+    static GMatDesc outMeta(GMatDesc ly, GMatDesc ry,
+                            const cv::gapi::ster::OUTPUT_FORMAT,
```

Comment:
Should we also check that OUTPUT_FORMAT is one of existent?

Reply:
Done